samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst

```diff
@@ -0,0 +1,351 @@
+
+.. _nrf_cloud_mqtt_multi_services_sample_temperature:
+.. _nrf_cloud_mqtt_multi_services_sample_temperature_sensing:
+
+Temperature Sensing
+===================
+
+Temperature sensing is almost entirely implemented in :file:`src/temperature.c`.
+This includes generation of false temperature readings on the `Nordic nRF9160 DK`_, which does not have a built-in temperature sensor.
+
+However, use of the built-in temperature sensor of the `Nordic Thingy:91`_ requires a `devicetree overlay <Zephyr Devicetree Overlays_>`_, namely :file:`boards/thingy91_nrf9160_ns.overlay`, as well as a couple of enabled kconfig options (:kconfig:option:`CONFIG_SENSOR` and :kconfig:option:`CONFIG_BME680`).
+The devicetree overlay is automatically applied during compilation whenever the ``thingy91_nrf9160_ns`` boardname is selected.
+The required kconfig options are implicitly enabled by :ref:`CONFIG_TEMP_DATA_USE_SENSOR <CONFIG_TEMP_DATA_USE_SENSOR>`.
+
+.. _nrf_cloud_mqtt_multi_services_sample_location_tracking:
+.. _nrf_cloud_mqtt_multi_services_sample_location:
+
+Location Tracking
+=================
+
+All matters concerning location tracking are handled in :file:`src/location_tracking.c`.
+For the most part, this amounts to setting up a periodic location request, and then passing the results to a callback configured by :file:`src/application.c`.
+However, there are a couple of other tasks which must additionally be implemented:
+
+Firstly, the :ref:`location library <lib_location>` requires assisted GPS (AGPS) data in order have the fastest possible `time-to-first-fix <TTFF_>`_.
+The :ref:`location library <lib_location>` requests this data automatically, but has no way to automatically receive the response.
+Therefore, all MQTT messages received from `nRF Cloud`_ must be be passed directly through to the :ref:`nRF Cloud AGPS library <lib_nrf_cloud_agps>` to be checked for AGPS data.
+The :func:`location_agps_data_handler` function handles this task, and is sent MQTT messages directly from :file:`src/connection.c`
+
+Secondly, the GNSS module cannot obtain a lock without the antenna first being properly configured.
+This configuration is performed by :func:`gnss_antenna_configure`.
+
+.. _nrf_cloud_mqtt_multi_services_sample_test_counter:
+
+Test Counter
+============
+
+A test counter may optionally be enabled by enabling :ref:`CONFIG_TEST_COUNTER <CONFIG_TEST_COUNTER>`.
+Every time a sensor sample is sent, this counter is incremented by one, and its current value is sent to `nRF Cloud`_.
+The value of the counter over time is automatically graphed in the `nRF Cloud`_ portal.
+This counter is useful for tracking, visualizing, and debugging connection loss, reset, and reestablishment behavior.
+
+
+.. _nrf_cloud_mqtt_multi_services_sample_device_message_formatting:
+
+Device Message Formatting
+=========================
+
+This sample manually constructs JSON-based `device messages <nRF Cloud Device Messages_>`_ using `cJSON`_.
```

Comment:
"Manually" seems out of place to me.  It would be manual if the code used snprintf to build the JSON, but instead, it uses the cJSON library.  Or maybe I'm missing what "automatically" or "implicitly" would mean in this case?

Reply:
Fair -- arguably, "automatic" would involve some use of the `nrf_codec` functions instead of directly constructing cJSON messages, but I agree it makes more sense just saying "constructs" instead of "manually constructs"